clouddrift/signal.py

```diff
@@ -41,57 +43,84 @@ def analytic_transform(
     >>> zp = analytic_transform(z)
     >>> zn = analytic_transform(np.conj(z))
 
-    To specify that a periodic boundary condition should be used
+    To specify that a periodic boundary condition should be used:
     >>> x = np.random.rand(99)
     >>> z = analytic_transform(x,boundary="periodic")
 
+    To specify that the time axis is along the first axis, and apply
+    zero boundary conditions:
+    >>> x = np.random.rand(100,99)
+    >>> z = analytic_transform(x,time_axis=0,boundary="zeros")
+
     Raises
     ------
     ValueError
         If ``boundary not in ["mirror", "zeros", "periodic"]``.
     """
-    # assume unidimensional input; add dimension option
-    m0 = len(x)
+    # time_axis must be in valid range
+    if time_axis < -1 or time_axis > len(x.shape) - 1:
+        raise ValueError(
+            f"time_axis ({time_axis}) is outside of the valid range ([-1,"
+            f" {len(x.shape) - 1}])."
+        )
+
+    # Reshape the inputs to ensure the time axis is last (fast-varying)
+    # should we add a condition to skip if time_axis=-1 or time_axis == len(x.shape)-1
+    # do we need to copy x here? does it matter for memory and performance?
```

Comment:
No. From `swapaxes` docs:

>        For NumPy >= 1.10.0, if `a` is an ndarray, then a view of `a` is
>        returned; otherwise a new array is created. For earlier NumPy
>        versions a view of `a` is returned only if the order of the
>        axes is changed, otherwise the input array is returned.

When a view is returned, it's the same data in memory, i.e. there's no copy. I will modify the comment to clarify this.